### ISSUE TITLE
Failed to build by docker

### ISSUE THREAD

**Issue by BrikeX** [microsoft/LightGBM#6420]
Running the following script:
```bash
mkdir lightgbm-docker
cd lightgbm-docker
wget https://raw.githubusercontent.com/Microsoft/LightGBM/master/docker/dockerfile-cli
docker build \
    -t lightgbm-cli \
    -f dockerfile-cli \
    .
```
then it reports:
```bash
ERROR: failed to solve: process “/bin/sh -c git clone --recursive --branch stable --depth 1 https://github.com/Microsoft/LightGBM && cd ./Lightgbm && cmake -B build -S . && cmake --build build -j4 && cmake --install build && cd “${HOME}” && rm -rf LightGBM” did not complete successfully: exit code: 2
```
It seems has some changes in the commit:
https://github.com/microsoft/LightGBM/commit/4332b9d5152c90b211e6667093c6c2bdda3ea15c

**Comment by jameslamb** [comment]
Thanks for the report.

Are you interested in investigating this and proposing a fix?

**Comment by BrikeX** [comment]
@jameslamb Has this commit https://github.com/microsoft/LightGBM/pull/6426 fixed this bug? I found that it could be fixed if I change the Ubuntu version to 22.04 and change the directory to "./LightGBM".

**Comment by jameslamb** [comment]
It will, yes, but it isn't merged yet.
